docs/book/src/user/troubleshooting.md

```diff
@@ -1,5 +1,23 @@
 # Troubleshooting
 
+## Node bootstrap failures
+
+Failures during Node bootstrapping can have a lot of different causes. For example, ClusterAPI resources might be 
+misconfigured or there might be problems with the network. The following steps describe how bootstrap failures can 
+be troubleshooted systematically.
+
+1. Access the Node via ssh.
+1. Take a look at cloud-init logs via `less /var/log/cloud-init-output.log` or `journalctl -u cloud-init --since "1 day ago"`.
+   (Note: cloud-init persists logs of the commands it executes (like kubeadm) only after they have returned.)
+1. It might also be helpful to take a look at `journalctl --since "1 day ago"`.
+1. If you see that kubeadm times out waiting for the static Pods to come up, take a look at:
+   1. containerd: `crictl ps -a`, `crictl logs`, `journalctl -u containerd`
+   1. Kubelet: `journalctl -u kubelet --since "1 day ago"`
+      (Note: it might be helpful to increase the Kubelet log level by e.g. setting `--v=8` via 
+      `systemctl edit --full kubelet && systemctl restart kubelet`)
+1. If Node bootstrapping consistently fails and the kubeadm logs are not verbose enough, the `kubeadm` verbosity 
```

Comment:
this section assumes cloud-init and kubeadm which does not apply to all bootstrap providers, maybe add something like "when using [CABPK](https://github.com/kubernetes-sigs/cluster-api/tree/master/bootstrap/kubeadm)" at the top or as a separate header